### ISSUE TITLE
NullReferenceException in WaitForDeviceStreamRequestAsync

### ISSUE THREAD

**Issue by varunpuranik** [Azure/azure-iot-sdk-csharp#780]
## Description of the issue:
I am getting a NullReferenceException on the client side when waiting for Device Stream request - 

## Code sample exhibiting the issue:
```
Console.WriteLine($"Waiting for stream request...");
DeviceStreamRequest streamRequest = await this.deviceClient.WaitForDeviceStreamRequestAsync(cts).ConfigureAwait(false);
Console.WriteLine("Received stream request");
if (streamRequest != null)
{
    Task streamTask = this.StartStream(streamRequest);
}
```

## Console log of the issue:
Here is the call stack - 
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Azure.Devices.Client.Transport.AmqpTransportHandler.WaitForDeviceStreamRequestAsync(CancellationToken cancellationToken) in E:\workspace\azure-iot-sdk-csharp\iothub\device\src\Transport\AmqpTransportHandler.cs:line 283
   at Microsoft.Azure.Devices.Client.Transport.ErrorDelegatingHandler.ExecuteWithErrorHandlingAsync[T](Func`1 asyncOperation) in E:\workspace\azure-iot-sdk-csharp\iothub\device\src\Transport\ErrorDelegatingHandler.cs:line 190
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.<>c__DisplayClass25_0.<<WaitForDeviceStreamRequestAsync>b__0>d.MoveNext() in E:\workspace\azure-iot-sdk-csharp\iothub\device\src\Transport\RetryDelegatingHandler.cs:line 254
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.WaitForDeviceStreamRequestAsync(CancellationToken cancellationToken) in E:\workspace\azure-iot-sdk-csharp\iothub\device\src\Transport\RetryDelegatingHandler.cs:line 251
   at Microsoft.Azure.Devices.Client.InternalClient.WaitForDeviceStreamRequestAsync(CancellationToken cancellationToken) in E:\workspace\azure-iot-sdk-csharp\iothub\device\src\InternalClient.cs:line 930
   at Microsoft.Azure.Devices.Edge.Agent.IoTHub.LogsClient.InitLogsStreaming() in E:\iot\g7\iotedge\edge-agent\src\Microsoft.Azure.Devices.Edge.Agent.IoTHub\LogsClient.cs:line 53
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Azure.Devices.Client.Transport.AmqpTransportHandler.WaitForDeviceStreamRequestAsync(CancellationToken cancellationToken) in E:\workspace\azure-iot-sdk-csharp\iothub\device\src\Transport\AmqpTransportHandler.cs:line 283
   at Microsoft.Azure.Devices.Client.Transport.ErrorDelegatingHandler.ExecuteWithErrorHandlingAsync[T](Func`1 asyncOperation) in E:\workspace\azure-iot-sdk-csharp\iothub\device\src\Transport\ErrorDelegatingHandler.cs:line 190
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.<>c__DisplayClass25_0.<<WaitForDeviceStreamRequestAsync>b__0>d.MoveNext() in E:\workspace\azure-iot-sdk-csharp\iothub\device\src\Transport\RetryDelegatingHandler.cs:line 254
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.WaitForDeviceStreamRequestAsync(CancellationToken cancellationToken) in E:\workspace\azure-iot-sdk-csharp\iothub\device\src\Transport\RetryDelegatingHandler.cs:line 251
   at Microsoft.Azure.Devices.Client.InternalClient.WaitForDeviceStreamRequestAsync(CancellationToken cancellationToken) in E:\workspace\azure-iot-sdk-csharp\iothub\device\src\InternalClient.cs:line 930
   at Microsoft.Azure.Devices.Edge.Agent.IoTHub.LogsClient.InitLogsStreaming() in E:\iot\g7\iotedge\edge-agent\src\Microsoft.Azure.Devices.Edge.Agent.IoTHub\LogsClient.cs:line 53

**Comment by Boltac** [comment]
Getting the same error exactly after 60 seconds

**Comment by dsajanice** [comment]
Seeing the same issue.

```
            Console.WriteLine($"Before: {DateTime.Now.ToString()}");
            try {
                await deviceClient.WaitForDeviceStreamRequestAsync();
            } 
            catch (Exception e)
            {
                Console.WriteLine($"After: {DateTime.Now.ToString()}");
                Console.WriteLine($"Exception: {e.ToString()}");
            }
```
Output:

Before: 2/8/2019 10:53:44 AM
After: 2/8/2019 10:54:46 AM
Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Azure.Devices.Client.Transport.AmqpTransportHandler.WaitForDeviceStreamRequestAsync(CancellationToken cancellationToken)
   at Microsoft.Azure.Devices.Client.Transport.ErrorDelegatingHandler.ExecuteWithErrorHandlingAsync[T](Func`1 asyncOperation)
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.<>c__DisplayClass25_0.<<WaitForDeviceStreamRequestAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.WaitForDeviceStreamRequestAsync(CancellationToken cancellationToken)
   at Microsoft.Azure.Devices.Client.InternalClient.WaitForDeviceStreamRequestAsync(CancellationToken cancellationToken)
   at Device.DeviceRegistration.RegisterDevice(IConfigurationRoot configuration) in C:\Users\jadsa\Repos\azure-iot-gateway-updator\csharp\Device\Program.cs:line 47

**Comment by prmathur-microsoft** [comment]
@ewertons is this resolved in your PR?

**Comment by jenoola** [comment]
Fix checked in the new release 1.20.0

**Comment by az-iot-builder-01** [comment]
@Boltac, @dsajanice, @prmathur-microsoft, @jenoola, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
